### ISSUE TITLE
Handling candidate changes in a dictionary?

### ISSUE THREAD

**Issue by rtoy** [tabatkins/bikeshed#2109]
An upcoming change in WebAudio wants to change the dictionary defined in bikeshed by
```
<pre class="idl">
dictionary AudioContextOptions {
	(AudioContextLatencyCategory or double) latencyHint = "interactive";
	float sampleRate;
};
</pre>
```
to add a new attribute like so:
```
<div class="correction" id="a13-2">
  <span class="marker">Candidate Addition
    <a href="https://github.com/WebAudio/web-audio-api-v2/issues/13">V2 Issue 13-1.</a>
  </span>
  Changes

<pre class="idl">
dictionary AudioContextOptions {
	(AudioContextLatencyCategory or double) latencyHint = "interactive";
	float sampleRate;
        <ins cite=#a13-2>
	(AudioContextRenderSizeCategory or unsigned long) renderSizeHint = "default";
        </ins>
};
</pre>
</div>
```
Ideally, we want to add `<ins>` and `</ins>` around the new attribute to show that it's the new stuff.

Not really sure how to do this.  Currently it looks as if bikeshed strips off the `<ins>`.

**Comment by tabatkins** [comment]
Yes, Bikeshed's IDL parsing/marking-up treats the contents of the block as plain text, rather than attempting to merge with the existing markup.

*In general*, merging two versions of markup is impossible; Bikeshed currently does so when adding highlighting, because the highlighting is effectively a character-level modification and so any highlighting markup can be safely broken up into separate elements when links/dfns intersect with it. But for arbitrary markup, there are cases that can't be merged.

That said, it's *likely* that markup can be safely merged with IDL links/dfns, because those links/dfns are fairly tightly scoped and it's not going to be common to have markup crossing those boundaries. So this could be fixed, with the possibility that Bikeshed complains it can't merge and just goes with one or the other.

For now, tho, I recommend just wrapping the entire `pre` in ins/del, because there's no way at the moment to do this.

**Comment by rtoy** [comment]
If you mean something like:
```
<div class="correction" id="a13-2">
  <span class="marker">Candidate Addition
    <a href="https://github.com/WebAudio/web-audio-api-v2/issues/13">V2 Issue 13-1.</a>
  </span>
  Changes
<del>
<pre class="idl">
dictionary AudioContextOptions {
  oldstuff;
};
</pre>
</del>
<ins>
<pre class="idl">
dictionary AudioContextOptions {
  newstuff;
};
</pre>
</ins>
</div>
```
then this doesn't work either because bikeshed then complains about multiple dictionary dfns linking the same AudioContextOptions, including duplicated dictionary members.  I guess this is actually true.

However, the generated result looks ok.  And pressing on a link to AudioContextOptions takes me to the first dfn, which is good enough.  See https://pr-preview.s3.amazonaws.com/rtoy/web-audio-api/pull/2369.html#a13-2

**Comment by svgeesus** [comment]
> For now, tho, I recommend just wrapping the entire pre in ins/del, because there's no way at the moment to do this.

That should work: `<ins>` and `<del>` take [phrasing content and flow content](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/del) so `<pre>` inside `<ins>`/`<del>` would be accepted. 

Note that `<table>` is [flow content](https://developer.mozilla.org/en-US/docs/Web/Guide/HTML/Content_categories#flow_content) but `<tr>` and `<td>` are not; so if a table is changed and the structure is identical,  `<ins>`/`<del>`  can be used inside individual `<td>`s otherwise the entire table needs to be duplicated (and the IDs de-duplicated, which is messy with internal links).

**Comment by tabatkins** [comment]
I meant "there's no way" to have Bikeshed merge ins/del markup with widlparser markup. Putting the entire pre block inside of ins/del is of course possible, which is why I suggested it. ^_^

(I'm nearly done with the patch allowing merging the markups, tho it still runs into issues with widlparser itself seeing the text contents of both elements. I'll need to do more work to make that reasonable.)

**Comment by fantasai** [comment]
@rtoy What I usually end up doing in the duplicate dfn case is to make the deleted instance not a dfn. (I'm frequently putting diffs into our changelogs, so I'm familiar with the problem.)
